### PR TITLE
Add docstrings for all commands

### PR DESCRIPTION
**Pull request by ararslan**
This is part of @tkelman's declared request for moar docs noted in JuliaLang/METADATA.jl#5589.

### CODE REVIEW

File: src/Git.jl [hunk 1]
```diff
@@ -75,6 +97,11 @@ immutable State
     work::Compat.UTF8String
 end
 
+"""
+    Git.snapshot(; dir="")
+
+Return a `State` object that consisting of a snapshot of the given repository.
```
Comment (tkelman) [inline src/Git.jl]: that captures a snapshot ?

File: src/Git.jl [hunk 2]
```diff
@@ -12,6 +12,12 @@ depsjl = joinpath(dirname(@__FILE__), "..", "deps", "deps.jl")
 isfile(depsjl) ? include(depsjl) : error("Git.jl not properly installed. " *
     "Please run\nPkg.build(\"Git\")")
 
+"""
+    Git.dir([d])
```
Comment (tkelman) [inline src/Git.jl]: looks like the argument is needed?

File: src/Git.jl [hunk 3]
```diff
@@ -90,6 +117,11 @@ function snapshot(; dir="")
     State(head, index, work)
 end
 
+"""
+    Git.restore(s; dir="")
```
Comment (tkelman) [inline src/Git.jl]: `s::State`

**Comment by tkelman** [comment]
great, thank you! is the wip that you plan on adding more or ?

**Comment by ararslan** [comment]
Yes, WIP because I plan to document *all* the things.

<a href="https://imgflip.com/i/17apoe"></a>

File: src/Git.jl (hunk 2 above)
Comment (ararslan) [inline src/Git.jl]: Oh yeah, my bad. Was looking at `git`, which doesn't require the argument.

**Comment by ararslan** [comment]
I think I finished getting all of the docstrings in there. Is there any other kind of documentation that would be nice to have here, be it in this PR or in general?

File: src/Git.jl [hunk 4]
```diff
@@ -24,6 +24,12 @@ function dir(d)
     normpath(d, Base.readchomp(setenv(`$gitcmd rev-parse --git-dir`, dir=d)))
 end
 
+"""
+    Git.git([d])
+
+Return a Git command that refers to the work tree and directory given by `d`, or the
+current work tree and directory of `d` is not specified.
```
Comment (tkelman) [inline src/Git.jl]: if `d` is not specified

**Comment by tkelman** [comment]
This is perfect. If Documenter.jl is a quick thing to set up for populating api docs into the readme then I should figure out how to use it at some point, but this package is probably not complicated or important enough to be worth a lot of effort there.

**Comment by ararslan** [comment]
Ah yeah, I don't know how to use Documenter either. I should also figure it out because I think there's some stuff around that still uses Docile, as well as things that could generally use some nice docs.

Is documentation here the only thing that's holding up JuliaLang/METADATA.jl#5589?

**Comment by tkelman** [comment]
I was just waiting the customary 24 hour feedback period for new package registrations. Docs can be a future tag.

**Pull request merged**